corsheaders/middleware.py

```diff
@@ -103,7 +103,7 @@ def process_response(self, request, response):
         Add the respective CORS headers
         """
         origin = request.META.get('HTTP_ORIGIN')
-        if request._cors_enabled and origin:
+        if getattr(request, '_cors_enabled', False) and origin:
```

Comment:
This would make (in the short circuit scenario) the CORS middleware add CORS headers to the response in projects that use MIDDLEWARE_CLASSES and not add the CORS headers to response in projects that use MIDDLEWARE Django setting.

My thinking here was: Since CORS middleware is MIDDLEWARE compatible it makes sense for it to behave consistently in projects that use MIDDLEWARE and MIDDLEWARE_CLASSES.

Is your argument:
- "make it behave differently for MIDDLEWARE and MIDDLEWARE_CLASSES because when MIDDLEWARE_CLASSES is used then the user explicitly opts in to the "old" middleware mechanism"?

or
- "make it always return CORS headers regardless of which middleware setting is used, even if there's a middleware in front of CORS middleware that returns a response during process_request"?

If it's the latter, then I'm not sure it's possible with the new Django middleware mechanism.